testbed/testbed/validator.go

```diff
@@ -0,0 +1,312 @@
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: recdSpan.Name,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+		return
+	}
+	if hex.EncodeToString(sentSpan.TraceId) != hex.EncodeToString(recdSpan.TraceId) {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "TraceId",
+			expectedValue: hex.EncodeToString(sentSpan.TraceId),
+			actualValue:   hex.EncodeToString(recdSpan.TraceId),
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if hex.EncodeToString(sentSpan.SpanId) != hex.EncodeToString(recdSpan.SpanId) {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "SpanId",
+			expectedValue: hex.EncodeToString(sentSpan.SpanId),
+			actualValue:   hex.EncodeToString(recdSpan.SpanId),
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.TraceState != recdSpan.TraceState {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "TraceState",
+			expectedValue: sentSpan.TraceState,
+			actualValue:   recdSpan.TraceState,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if hex.EncodeToString(sentSpan.ParentSpanId) != hex.EncodeToString(recdSpan.ParentSpanId) {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "ParentSpanId",
+			expectedValue: hex.EncodeToString(sentSpan.ParentSpanId),
+			actualValue:   hex.EncodeToString(recdSpan.ParentSpanId),
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.Name != recdSpan.Name {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "Name",
+			expectedValue: sentSpan.Name,
+			actualValue:   recdSpan.Name,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.Kind != recdSpan.Kind {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "Kind",
+			expectedValue: sentSpan.Kind,
+			actualValue:   recdSpan.Kind,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.StartTimeUnixNano != recdSpan.StartTimeUnixNano {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "StartTimeUnixNano",
+			expectedValue: sentSpan.StartTimeUnixNano,
+			actualValue:   recdSpan.StartTimeUnixNano,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if sentSpan.EndTimeUnixNano != recdSpan.EndTimeUnixNano {
+		af := &AssertionFailure{
+			typeName:      "Span",
+			dataComboName: sentSpan.Name,
+			fieldPath:     "StartTimeUnixNano",
+			expectedValue: sentSpan.EndTimeUnixNano,
+			actualValue:   recdSpan.EndTimeUnixNano,
+		}
+		v.assertionFailures = append(v.assertionFailures, af)
+	}
+	if len(sentSpan.Attributes) != len(recdSpan.Attributes) {
```

Comment:
Can you add a TODO to also compare attribute key-values not just their count?

Reply:
Done